platform/core.multitabs/src/org/netbeans/core/multitabs/impl/DocumentSwitcherTable.java

```diff
@@ -152,7 +152,21 @@ boolean onMouseEvent( MouseEvent e ) {
                         TabActionEvent tae = new TabActionEvent( this, TabbedContainer.COMMAND_CLOSE, tabIndex );
                         controller.postActionEvent( tae );
                         return true;
+                    }else if(tabIndex == -1){
+                        List<TabData> tabs = controller.getTabModel().getTabs();
+                        ProjectProxy project = item.project;
+                        ProjectSupport projectSupport = ProjectSupport.getDefault();
+                        for(int i = tabs.size(); i-- > 0;){
+                            TabData projectTab = tabs.get(i);
+                            if(projectSupport.getProjectForTab(projectTab) == (project) || projectSupport.getProjectForTab(projectTab).equals(project)){
```

Comment:
the equals check alone would be sufficient. but to be sure, lets add a null check too. We can also use the opportunity to store the result in a variable so that we call the method only once (it could be expensive if the result is not cached).
```java
  ProjectProxy projectForTab = projectSupport.getProjectForTab(projectTab);
  if (projectForTab != null && projectForTab.equals(project)) {
```

Reply:
The reason I am not checking for null, is because there is a case, when no project is associated when file and variables projectForTab and project are null. 

Should I add another `if`  for that case?

Reply:
good thinking. How about:
```java
  ProjectProxy projectForTab = projectSupport.getProjectForTab(projectTab);
  if ((project == null && projectForTab == null) || (projectForTab != null && projectForTab.equals(project))) {
```

Reply:
That should work, gonna try it. Is it okay to have such a long condition?

Reply:
sure. Its not a problem as long its somewhat readable. Your version was fine too, but without knowing about the special case where both are null, it almost looks like a bug. Making it explicit answers possible questions right away ;)